### PR TITLE
Type AppContext

### PR DESCRIPTION
**Pull request by OzzieOrca**

### CODE REVIEW

File: src/appContext.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
 
-const AppContext = React.createContext({});
+export const appContextDefaultValue = { orgId: '' };
 
-export default AppContext;
+export type AppContextValue = typeof appContextDefaultValue;
```
Comment (TheNoodleMoose) [inline src/appContext.ts]: So does this type inherit and know that appContextDefaultValue is equal to a string? Therefore, making the exported `type AppContextValue` be a string? I thought we had to define what type it is unless TypeScript can infer that for us.

File: src/testUtils/index.tsx [hunk 2]
```diff
@@ -1,30 +1,42 @@
...
+        mocks: mocks = {},
+        appContext = appContextDefaultValue,
+    }: RenderWithContextParams = {},
 ) => {
     const mockApolloClient = createApolloMockClient(mocks);
 
     // Warning: don't call any functions in here that return new instances on every call.
     // All the props need to stay the same otherwise renderer won't work.
-    const wrapper = ({ children }: { children: ReactNode }) => (
+    const wrapper = ({ children }: { children?: ReactNode }) => (
```
[the remark this answers is not in the document]
Comment (OzzieOrca) [reply src/testUtils/index.tsx]: Ya it means optional. I was getting an error without it. I guess the wrapper component could get called with no children if you rendered null or something... Thought it was a little weird too.

File: src/appContext.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
 
-const AppContext = React.createContext({});
+export const appContextDefaultValue = { orgId: '' };
```
[the remark this answers is not in the document]
Comment (OzzieOrca) [reply src/appContext.ts]: Ya it starts with having `orgId` defined but as an empty string. I suppose `orgId` could have been optional but we need it for all the code Bartosz wrote. Eventually we'll have to use the router for this as we use React in more places.

File: src/appContext.ts (hunk 1 above)
Comment (OzzieOrca) [reply src/appContext.ts]: Ya `typeof` generates a type based on the variable which is an object with `orgId` having a string value. I could have hardcoded it but this seemed nicer.
Screen Shot 2019-09-05 at 12 35 43 PM
TypeScript can infer a lot which is nice. It should yell at you if it doesn't know the type of something.

File: src/communityInsights/containers/StepsOfFaithPage/index.tsx [hunk 4]
```diff
@@ -20,7 +20,7 @@ import {
 
 const StepsOfFaithPage = () => {
     const { t } = useTranslation('insights');
-    const { orgId }: any = useContext(AppContext);
+    const { orgId } = useContext(AppContext);
```
[the remark this answers is not in the document]
Comment (OzzieOrca) [reply src/communityInsights/containers/StepsOfFaithPage/index.tsx]: I think the default value is only used if there was no provider. https://github.com/CruGlobal/missionhub-web/blob/95bbb570a99b1859315dd4bb00e8a5032cc035c9/src/communityInsights/insights.tsx#L22 gives the context a real value. Context is a nice way to share a value between components without having to pass it down through every component in the tree.

**Pull request merged**